### PR TITLE
Only try to get a compilation for a language that supports it.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/Microsoft/visualfsharp/pull/3499

I would recommend reviewing this as: https://github.com/dotnet/roslyn/pull/21721/files?w=1 as there is an indentation change here.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide @jasonmalinowski

File: src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentProjectsFinder.cs [hunk 1]
```diff
@@ -299,7 +299,7 @@ private static bool IsInternalsVisibleToAttribute(AttributeData attr)
 
             var internalsVisibleToMap = CreateInternalsVisibleToMap(sourceAssembly);
 
-            SymbolKey? sourceAssemblySymbolKey = null;
+            var sourceAssemblySymbolKey = sourceAssembly.GetSymbolKey();
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentProjectsFinder.cs]: these symbol keys are not expensive to make.  I'm just removing this premature optimization where we attempt to delay create these.

**Comment by CyrusNajmabadi** [comment]
I would recommend reviewing this as: https://github.com/dotnet/roslyn/pull/21721/files?w=1

**Comment by CyrusNajmabadi** [comment]
retest  ubuntu_16_debug_prtest please

**Comment by dsyme** [comment]
@CyrusNajmabadi @jasonmalinowski Thanks for looking after this.  And particularly thanks for adding a test to help pin down the behaviour.

**Comment by cartermp** [comment]
Will this be for 15.5?

**Comment by CyrusNajmabadi** [comment]
@cartermp Yes.  I believe roslyn master if 15.5.  @Pilchie can confirm.

**Comment by CyrusNajmabadi** [comment]
Note: i cannot guarantee that this will address all issues.  My guess is that we'll need to do a little back/forth to squash all the issues.  When we did this with TypeScript it only took a few PRs to get through everything, so i'm not too concerned about a long bug tail here.

**Comment by jasonmalinowski** [comment]
@cartermp Yes, this is 15.5.

**Pull request merged**